Sources/MapboxMaps/Foundation/MapView.swift

```diff
@@ -574,7 +578,9 @@ open class MapView: UIView {
 
         if needsDisplayRefresh {
             needsDisplayRefresh = false
+            metricsReporter?.beforeMetalViewDrawCallback(metalView: metalView)
             metalView?.draw()
+            metricsReporter?.afterMetalViewDrawCallback(metalView: metalView)
```

Comment:
I wonder if it would make sense to use compilation conditions for that, like
```swift
#if METRICS        
        metricsReporter?.beforeDisplayLinkCallback(displayLink: displayLink)
         defer { metricsReporter?.afterDisplayLinkCallback(displayLink: displayLink) }
#endif
```
Maybe I’m being paranoid but this way we would be sure that measurement code won’t have any effect on regular usage.

Reply:
It was a great idea, but unfortunately, I cannot make it work  :( 
<img width="1206" alt="image" src="https://user-images.githubusercontent.com/735178/186460924-64c3013b-f76b-4fb7-82da-4846180de38a.png">